cutting down (#98)
jaybee84/ml-in-rd: content/05.prior-knowledge.md

A potentially powerful strategy for evaluating genotype-phenotype relationships or repurposing drugs when large datasets are scarce is to use knowledge graphs.
Knowledge graphs integrate related-but-different data types, creating a rich data source. 
Examples of public biomedical knowledge graphs and frameworks that could be useful in rare disease include the Monarch Graph Database[@doi:10.1093/nar/gkw1128], hetionet[@doi:10.7554/eLife.26726], PheKnowLator[@doi:10.1101/2020.04.30.071407], and the Global Network of Biomedical Relationships[@doi:10.1093/bioinformatics/bty114]. 
These graphs connect information like genetic, functional, chemical, clinical, and ontological data to enable the exploration of relationships of data with disease phenotypes through manual review[@doi:10.1093/database/baaa015] or computational methods[@doi:10.1101/727925; @doi:10.1186/s12911-019-0938-1]. 

In the academic rare disease space, there a few pioneering examples of ML-based mining of knowledge graphs to repurpose drugs[@doi:10.1101/727925] and classify rare diseases[@doi:10.1186/s12911-019-0938-1].

> These studies make it clear that there are challenges in using machine learning using graph databases in rare disease.

Reviewer: ```suggestion
These studies make it clear that there are challenges in using machine learning based on knowledge graphs in rare disease.
```

Reviewer: distinguishing our graphs from graph databases like https://aws.amazon.com/neptune/, https://docs.microsoft.com/en-us/azure/cosmos-db/graph-introduction

Author: Fair point for clarity!

For example, these projects rely on a gold standard dataset to validate the performance of the models; often, there are not robust gold standard datasets available for individual rare diseases.
They also evaluate rare diseases in an unbiased manner, rather than interrogating a specific disease of interest.
Consequently, it is not yet clear how effective these approaches, and knowledge graphs in general, are in studying a specific disease of interest; more work needs to be done to identify methods that can provide actionable insights for a specific rare disease application. 

Beyond the aforementioned studies, there are not many examples of studies in the public domain that leverage knowledge graphs to characterize rare diseases. 

> Private entities (e.g. healx, Boehringer Ingelheim), however, are performing an undisclosed amount of work to create proprietary rare disease knowledge graphs for ML-based drug discovery applications.

Reviewer: any particular reason to remove drugbankplus? Just curious. :)

Author: To remove one word, but also I thought it was more accurate - DrugBankPlus is (at least one of the) data sources for Healx's project (https://www.drugbankplus.com/downloads/DrugBankHealx_CaseStudy.pdf) so it's really a duplicate rather than an independent effort

The existence of private companies pursuing this idea, as well as the availability of public biomedical knowledge graphs, suggests that this is likely a fruitful untapped area of rare disease research in the public arena. 
More work needs to be done to assess 1) which graphs and graph features capture the salient information about rare diseases, 2) the utility of ML methods to obtain actionable insights about rare diseases and 3) which problems - like drug discovery, identification of novel rare diseases, or assessment of genotype-phenotype relationships - can be interrogated using ML of knowledge graphs.

> #### Wisdom of the crowd: rare disease applications of ensemble methods

Reviewer: I think this part is better suited for the transition from model-complexity section to prior-knowledge section

Author: This isn't clear to me - do you mean we should talk about ensemble models before any of the individual methods?

Reviewer: My feeling is that the bit about ensembling is relating more closely to the present model-complexity section as a technique for stable feature selection, compared to prior-knowledge section.

Reviewer: similarly for the binary vs non-binary classification

Author: hmm, interesting. I'll file it as an issue!

Features learned from transcriptomic data could be coordinated sets of genes involved in a biological process that are descriptive in some way [@doi:10.1038/s41467-020-14666-6].

In the rare disease domain, Dincer et al. leveraged publicly available acute myeloid leukemia (AML) gene expression data to improve the prediction of _in vitro_ drug responses [@doi:10.1101/278739]. 

> The authors trained a variational autoencoder (an unsupervised neural network that learns a series of representations from data), or VAE, on AML data that had been collected over time without the phenotypic information they were interested in (drug response). 

Reviewer: ```suggestion
The authors trained a variational autoencoder (an unsupervised neural network that learns a series of representations from data), or VAE, on AML data that had been collected over time without the desired phenotypic information (drug response). 
```
